components/RoomScreen.js

```diff
@@ -13,47 +13,76 @@ import auth from '@react-native-firebase/auth';
 
 import Constants from '../config/constants';
 import * as GLOBAL from '../state/global';
-import { CoffeeCupSmall, UserImage, BackArrow } from './Images';
+import { CoffeeCupSmall, UserImage, BackArrow, LeaveRoom } from './Images';
 import PickingPhase from './Room-Components/PickingPhase';
 import ResultsPhase from './Room-Components/ResultsPhase';
 import UsersList from './Room-Components/UsersList';
+import constants from '../config/constants';
 
 class RoomScreen extends Component {
     state = {
         usersListOpen: false,
-        roomState: {"hostSocketID":"3dudDTtDNfa_4ooSAAAB","phase":"RESULTS_PHASE","voteByUserID":{"3dudDTtDNfa_4ooSAAAB":1,"jSDg2ki-eL9lA9NEAAAC":2,"WvJaSjDcLuxcLy2NAAAD":6,"fIYzU092AqYv1DOWAAAE":4},"connectedUsersByID":{"3dudDTtDNfa_4ooSAAAB":{"nickname":"FakeTanneryh","socketID":"3dudDTtDNfa_4ooSAAAB"},"jSDg2ki-eL9lA9NEAAAC":{"nickname":"Fa","socketID":"jSDg2ki-eL9lA9NEAAAC"},"WvJaSjDcLuxcLy2NAAAD":{"nickname":"Favncbv","socketID":"WvJaSjDcLuxcLy2NAAAD"},"fIYzU092AqYv1DOWAAAE":{"nickname":"Faghjfjjgfhj","socketID":"fIYzU092AqYv1DOWAAAE"}},"deck":[{"value":1,"tag":"ezz"},{"value":3,"tag":"hard"},{"value":8,"tag":"harderer"},{"value":100,"tag":"kill me"},{"value":999,"tag":"super hard"},{"value":-1,"tag":"done already"},{"tag":"Abstain","value":"ABSTAIN"}]},
+        roomState: null,
     };
 
     componentDidMount() {
-        if (!GLOBAL.roomServiceSocket) {
-            // This is an error state as it should always be initialized. We may want to just boot
-            // the user back to the home screen with a popup alert.
-            console.log("NO SOCKET ERROR STATE");
-        } else {
-            // We can add events like so:
-            GLOBAL.roomServiceSocket.on('room_state_changed', event => this.onRoomStateChanged(event));
-        }
+        this.props.navigation.addListener('focus', () => {
+            if (!GLOBAL.roomServiceSocket) {
+                console.log("NO SOCKET ERROR STATE");
+            } else {
+                // Add event listeners.
+                GLOBAL.roomServiceSocket.on('room_state_changed', event => this.onRoomStateChanged(event));
+                GLOBAL.roomServiceSocket.on('join_success', event => this.onJoinSuccess(event));
+                GLOBAL.roomServiceSocket.on('host_closed_connection', event => this.onHostClosedConnection(event));
+                // Fire off a join room request.
+                GLOBAL.roomServiceSocket.emit('join_room', { roomID: GLOBAL.roomName, nickname: GLOBAL.nickname });
+                GLOBAL.roomServiceSocket.emit('is_room_open', { roomID: GLOBAL.roomName });
+            }
+        });
+        this.props.navigation.addListener('blur', () => {
+            if (GLOBAL.roomServiceSocket) {
+                // We can "unsubscribe" to events like so (should be done for every listener that was initialized):
+                GLOBAL.roomServiceSocket.on('room_state_changed', event => { return; });

```

Comment:
This shouldn't actually unsubscribe: https://nodejs.org/api/events.html#events_emitter_on_eventname_listener
> Adds the listener function to the end of the listeners array for the event named eventName.

We spoke about this in discord though, just want to document this issue.